common-content/en/module/tools/installing-npm-dependencies/index.md

```diff
@@ -0,0 +1,124 @@
+
+```json
+{
+    "type": "module"
+}
+```
+
+The `package.json` contains a JSON object with information about your project. For now, we're just telling `npm` that our project is a module - this means we are allowed to use `import` in our program.
+
+From a terminal which is `cd`'d to the same directory as your `package.json` file, run `npm install commander`.
+
+This command does two things:
+1. Look in your `package.json` file - notice that now has a `dependencies` section listing `commander`. This means that if someone else downloads your program, they know they need to install `commander` to use it.
+2. There's now a `node_modules` directory alongside your `package.json`. Inside that is a directory named `commander` which contains the code for the `commander` library. This means `node` now knows how to find the code when you try to import it.
+
+{{<note type="Exercise">}}
+Try running your program again.
+
+What has changed since the last time you tried to run it (and it didn't work)?
+
+What has changed since the last time you successfully ran it?
+{{</note>}}
+
+Now that we have `commander` installed, let's try using it in our program:
+
+```js
+import { program } from "commander";
+import { promises as fs } from "node:fs";
+import process from "node:process";
+
+program
+    .name("count-containing-words")
+    .description("Counts words in a file that contain a particular character")
+    .option("-c, --char <char>", "The character to search for", "-");
+
+program.parse();
+
+const argv = program.args;
+if (argv.length != 1) {
+    console.error(`Expected exactly 1 argument (a path) to be passed but got ${argv.length}.`);
+    process.exit(1);
+}
+const path = argv[0];
+const char = program.opts().char;
+
+const content = await fs.readFile(path, "utf-8");
+const wordsContainingChar = content.split(" ").filter((word) => word.indexOf(char) > -1).length;
+console.log(wordsContainingChar);
+```
+
+{{<note type="Exercise">}}
+Try running this program with the `--help` flag.
+
+What do you see? Where do you think this behaviour and text came from?
+
+We didn't have to write all the code for this functionality - `commander` did most of it for us.
+{{</note>}}
+
+{{<note type="Exercise">}}
+Try running the program with different values of the `-c` flag. Try also specifying some other flags, like `--count`.
+
+Make sure you understand how it's behaving, and why.
+{{</note>}}
+
+Let's run through what we changed:
+
+```js
+program
+    .name("count-containing-words")
+    .description("Counts words in a file that contain a particular character")
+    .option("-c, --char <char>", "The character to search for", "-");
+```
+
+We told `commander` information about our program. We gave it a name, a description, and told it that it should allow a user to pass a flag name `-c` (or equivalently `--char`), and use a default value of `-` for that flag if it's not specified.
```

Comment:
I wonder if using hyphens is the best choice for this toy application - simply because here it could be confusing that we're using a hyphen as a value, whereas other places a hyphen is syntax.

Would switching to some other special character (`!` maybe?) be worth it?